packages/ws/src/utils/WorkerBootstrapper.ts

```diff
@@ -0,0 +1,160 @@
+import { isMainThread, parentPort, workerData } from 'node:worker_threads';
+import { Collection } from '@discordjs/collection';
+import type { Awaitable } from '@discordjs/util';
+import { WorkerContextFetchingStrategy } from '../strategies/context/WorkerContextFetchingStrategy.js';
+import {
+	WorkerRecievePayloadOp,
+	WorkerSendPayloadOp,
+	type WorkerData,
+	type WorkerRecievePayload,
+	type WorkerSendPayload,
+} from '../strategies/sharding/WorkerShardingStrategy.js';
+import type { WebSocketShardDestroyOptions } from '../ws/WebSocketShard.js';
+import { WebSocketShardEvents, WebSocketShard } from '../ws/WebSocketShard.js';
+
+/**
+ * Options for bootstrapping the worker
+ */
+export interface BootstrapOptions {
+	/**
+	 * Shard events to just arbitrarily forward to the parent thread for the manager to emit
+	 * Note: By default, this will include ALL events
+	 * you most likely want to handle dispatch within the worker itself
+	 */
+	forwardEvents?: WebSocketShardEvents[];
+	/**
+	 * Function to call when a shard is created for additional setup
+	 */
+	shardCallback?(shard: WebSocketShard): Awaitable<void>;
+}
+
+/**
+ * Utility class for bootstraping a worker thread to be used for sharding
+ */
+export class WorkerBootstrapper {
+	/**
+	 * The data passed to the worker thread
+	 */
+	protected readonly data = workerData as WorkerData;
+
+	/**
+	 * The shards that are managed by this worker
+	 */
+	protected readonly shards = new Collection<number, WebSocketShard>();
+
+	public constructor() {
+		if (isMainThread) {
+			throw new Error('Expected WorkerBootstrap to not be used within the main thread');
+		}
+	}
+
+	/**
+	 * Helper method to initiate a shard's connection process
+	 */
+	protected async connect(shardId: number): Promise<void> {
+		const shard = this.shards.get(shardId);
+		if (!shard) {
+			throw new Error(`Shard ${shardId} does not exist`);
```

Comment:
Shouldn't we use [`RangeError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/RangeError)?

Reply:
We already decided against that - this same check takes places in many other places. It doesn't make a lot of sense to use here, it's confusing if say, the manager in question is responsible for shards `1`, `3`, and `5` and, say, an op is attempted on shard `2`, that's not really a "range" error, IMO.

Reply:
> The `RangeError` object indicates an error when __a value is not in the set__ or range __of allowed values__.

While the class name is misleading, it's still talking about a value within the allowed values, whether or not that's within a range or within a set of ranges.